mopidy_spotify/playlists.py

```diff
@@ -52,19 +56,36 @@ def _get_playlist(self, uri, as_items=False):
     def refresh(self):
         if not self._backend._web_client.logged_in:
             return
+        elif self._refreshing:
+            return
 
-        logger.info("Refreshing Spotify playlists")
+        self._refreshing = True
 
-        with utils.time_logger("playlists.refresh()", logging.DEBUG):
-            _sp_links.clear()
-            self._backend._web_client.clear_cache()
-            count = 0
-            for playlist_ref in self._get_flattened_playlist_refs():
-                self._get_playlist(playlist_ref.uri)
-                count = count + 1
-            logger.info(f"Refreshed {count} Spotify playlists")
+        logger.info("Refreshing Spotify playlists")
 
-        self._loaded = True
+        def refresher():
+            try:
+                with utils.time_logger("playlists.refresh()", logging.DEBUG):
+                    _sp_links.clear()
+                    self._backend._web_client.clear_cache()
```

Comment:
What's the motivation for explicitly clearing cache here? Seem to come from 6dd39cf14252b517bc8fe972d0adf1d4f4d6bdb3 @kingosticks

Reply:
Because otherwise you might still be < max-age and therefore won't fetch the fresh data, possibly leaving you with stale data. Which you don't want because you specifically asked to refresh.

Reply:
Git it - my thinking is that the ideal behavior would be to always fetch data, but keep the cache until fresh data has arrived.

Does that make sense?